asserts/snap_asserts.go

```diff
@@ -698,3 +698,140 @@ func InitBuiltinBaseDeclaration(headers []byte) error {
 	builtinBaseDeclaration = a.(*BaseDeclaration)
 	return nil
 }
+
+type dateRange struct {
+	Since time.Time
+	Until time.Time
+}
+
+// SnapDeveloper holds a snap-developer assertion, defining the developers who
+// can collaborate on a snap while it's owned by a specific publisher.
+type SnapDeveloper struct {
+	assertionBase
+	developerRanges map[string][]*dateRange
+}
+
+// SnapID returns the snap id of the snap.
+func (snapdev *SnapDeveloper) SnapID() string {
+	return snapdev.HeaderString("snap-id")
+}
+
+// PublisherID returns the publisher's account id.
+func (snapdev *SnapDeveloper) PublisherID() string {
+	return snapdev.HeaderString("publisher-id")
+}
+
+func (snapdev *SnapDeveloper) checkConsistency(db RODatabase, acck *AccountKey) error {
+	// Check authority is the publisher or trusted.
+	authorityID := snapdev.AuthorityID()
+	publisherID := snapdev.PublisherID()
+	if !db.IsTrustedAccount(authorityID) && (publisherID != authorityID) {
+		return fmt.Errorf("snap-developer must be signed by the publisher or a trusted authority but got authority %q and publisher %q", authorityID, publisherID)
```

Comment:
In the context of validating the assertion, the error message makes sense to me as it specifically doesn't mention the snap-declaration, the *current* publisher or similar, and includes the authority and publisher IDs in the message. But I'd be happy to accept something better.

However, I completely agree the use of the assertion - that floating/orphaned is ok - is a bit surprising. Maybe mentioning this in the type's doc comment would help?